test/python/common.py

```diff
@@ -254,13 +252,48 @@ def slow_test(func):
     """
 
     @functools.wraps(func)
-    def _(*args, **kwargs):
-        if SKIP_SLOW_TESTS:
+    def _wrapper(*args, **kwargs):
+        if TEST_OPTIONS['skip_slow']:
             raise unittest.SkipTest('Skipping slow tests')
         return func(*args, **kwargs)
 
-    return _
+    return _wrapper
+
 
+def _add_credentials(args, kwargs):
```

Comment:
Would it make sense to tweak this function in order to provide more clarity and easier following of the logic as a whole? As things stand, I believe there are a couple of issues that are slightly convoluted (touching a bit on other comments by @delapuente, probably):
* this function seems to be called even if the mocking is enabled, which seems counter-intuitive (and with less importance, it is called unnecessarily as a side effect):
```python
    if _add_credentials(args, kwargs):
            if TEST_OPTIONS['rec'] or ****TEST_OPTIONS['mock_online']****:
                return VCR.use_cassette()(func)(*args, **kwargs)
            return func(*args, **kwargs)
```
* the fake credentials are appended in `requires_qe_access()` - making this function no longer the only place where credentials can be added.

What about making this function the "real only place where credentials for the tests are handled"? I'd personally also try to make return a dict with the credentials instead of a boolean, mostly for avoiding to modify an argument in-place and also for making it agnostic to kwargs, but not a super strong preference. This way it can be used as:

```
def requires_qe_access():
    ...
    kwargs.update(_add_credentials(mock_online))  # warning and exception raised inside
    ...
```

Which would in turn convert `requires_qe_access()` into mostly "decide which wrapper function should be returned based on the test options". As an aside, it might also imply reverting the name change for the `_add_credentials` function :grin:.